packages/vsce/src/utils/profileManagement.ts

```diff
@@ -52,273 +53,292 @@ export class ProfileManagement {
     return mProfileInfo;
   }
 
-  /**
-   * Makes axios GET request with path and axios config object given
-   * @param path
-   * @param config
-   * @returns
-   */
-  public static async makeRequest(path: string, config: AxiosRequestConfig) {
-    const response = await axios.get(path, config);
-    return response;
-  }
-
   public static cmciResponseXml2Json(data: string) {
     return JSON.parse(xml2json(data, { compact: true, spaces: 4 }));
   }
 
-  /**
-   * Populates the info
-   * @param profile
-   * @returns Array of type InfoLoaded
-   */
-  public static async getPlexInfo(profile: imperative.IProfileLoaded): Promise<InfoLoaded[]> {
-    const config: AxiosRequestConfig = {
-      baseURL: `${profile.profile.protocol}://${profile.profile.host}:${profile.profile.port}/CICSSystemManagement`,
-      auth: {
-        username: profile.profile.user,
-        password: profile.profile.password,
-      },
-    };
+  public static getSessionFromProfile(profile: imperative.IProfile): Session {
+    return new Session({
+      protocol: profile.protocol,
+      hostname: profile.host,
+      port: profile.port,
+      type: "basic",
+      user: profile.user,
+      password: profile.password,
+      rejectUnauthorized: 'rejectUnauthorized' in profile ? profile.rejectUnauthorized : true,
+    });
+  }
+
+  public static async regionIsGroup(session: Session, profile: imperative.IProfile): Promise<boolean> {
+
+    let checkIfSystemGroup: ICMCIApiResponse;
+    try {
+      checkIfSystemGroup = await getResource(session, {
+        name: "CICSRegionGroup",
+        cicsPlex: profile.cicsPlex,
+        regionName: profile.regionName,
+        criteria: `GROUP=${profile.regionName}`,
+      });
+    } catch (error) {
+      if (error instanceof imperative.ImperativeError) {
+        if (!error.mDetails.msg.toUpperCase().includes("NODATA")) {
+          throw error;
+        }
+      }
+    }
+
+    return checkIfSystemGroup?.response.resultsummary.recordcount !== "0";
```

Comment:
If we get an error (like NODATA), checkIfSystemGroup is still undefined, which means it isn't "0", which means we return true ... which is wrong.

I think the best thing to do is assign the recordcount to a variable and make sure it's defined. If it is, and it's non null - we got ourselves a result. Another option would be to parse it as an int and check it was greater than zero, but I suspect this will have its own issues.

```suggestion
    var recordCount = checkIfSystemGroup?.response.resultsummary.recordcount;
    return typeof recordCount !== "undefined" && recordCount !== "0";
```

Reply:
After a discussion with Andrew we came up with a clearer option... I'll suggest that separately.